src/components/graph/menus/create-node-menu.js

```diff
@@ -276,20 +316,63 @@ const CreateNodeMenu = ({
         [intl, activeNode?.type]
     );
 
+    const content = intl.formatMessage(
+        {
+            id:
+                nodeAction === NodeActions.REMOVE_SUBTREE
+                    ? 'deleteSubTreeConfirmation'
+                    : 'deleteNodeConfirmation',
+        },
+        {
+            nodeName: activeNode?.data?.label,
+            nodesNumber: getNodesFromSubTree(treeModel, activeNode?.id),
+        }
+    );
+
+    const handleOnValidate = useCallback(() => {
+        if (nodeAction === NodeActions.REMOVE_NODE) {
+            handleNodeRemoval(activeNode);
+        } else {
+            handleRemovalSubtree(activeNode);
+        }
+        handleClose();
+        setNodeAction(NodeActions.NO_ACTION);
```

Comment:
You can call handeOnClose instead of these 2 successive calls ...

Reply:
Indeed